---
title: C6318 | Microsoft Docs
ms.custom: ''
ms.date: 11/15/2016
ms.prod: visual-studio-dev14
ms.reviewer: ''
ms.suite: ''
ms.technology:
- vs-devops-test
ms.tgt_pltfrm: ''
ms.topic: article
f1_keywords:
- C6318
helpviewer_keywords:
- C6318
ms.assetid: 3284a83e-bb8e-461c-adcc-cfc66ceea05e
caps.latest.revision: 15
author: mikeblome
ms.author: mblome
manager: ghogen
ms.openlocfilehash: 58eb6f9f41eac282e2b7cce4477d204e88419152
ms.sourcegitcommit: af428c7ccd007e668ec0dd8697c88fc5d8bca1e2
ms.translationtype: MT
ms.contentlocale: fr-FR
ms.lasthandoff: 11/16/2018
ms.locfileid: "51770767"
---
# <a name="c6318"></a>C6318
[!INCLUDE[vs2017banner](../includes/vs2017banner.md)]

avertissement C6318 : __try mal définie /\__except : utilisation de la constante EXCEPTION_CONTINUE_SEARCH ou d’une autre constante égale à zéro dans l’expression de filtre d’exception. Le code dans le bloc de gestionnaire d’exceptions n’est pas exécuté.  
  
 Cet avertissement indique que si une exception se produit dans le bloc protégé de ce gestionnaire d’exceptions structuré, l’exception n’est pas gérée car la constante `EXCECPTION_CONTINUE_SEARCH` est utilisé dans l’expression de filtre d’exception.  
  
 Ce code est équivalent au bloc protégé sans le bloc de gestionnaire d’exceptions, car le bloc de gestionnaire n’est pas exécuté.  
  
## <a name="example"></a>Exemple  
 Le code suivant génère cet avertissement :  
  
```  
#include <excpt.h>  
#include <stdio.h>  
  
void f (char *pch)  
{  
  __try   
  {  
    // assignment might fail  
    *pch = 0;  
  }   
  __except (EXCEPTION_CONTINUE_SEARCH) // warning 6318  
  {  
    puts("Exception Occurred");  
  }  
}  
```  
  
 Pour corriger cet avertissement, utilisez le code suivant :  
  
```  
#include <excpt.h>  
#include <stdio.h>  
#include <windows.h>  
  
void f (char *pch)  
{  
  __try   
  {  
    // assignment might fail  
    *pch = 0;  
  }   
  __except( (GetExceptionCode() == EXCEPTION_ACCESS_VIOLATION) ?  
                EXCEPTION_EXECUTE_HANDLER : EXCEPTION_CONTINUE_SEARCH )  
  {  
    puts("Access violation");  
  }  
}   
```  
  
## <a name="see-also"></a>Voir aussi  
 [try-except, instruction](http://msdn.microsoft.com/library/30d60071-ea49-4bfb-a8e6-7a420de66381)



